index.js

```diff
@@ -266,3 +249,133 @@ function init() {
+	var arc = {};
+    for (var i = 0; i < 5; ++i) {
+        for (var c = 0; c < 8; ++c) {
+            if ($(colors[c] + '.' + i).checked) {
+                if (!(i in arc)) {
+                    arc[i] = [colors[c]];
+                } else {
+                    arc[i].push(colors[c]);
+                }
+            }
+        }
+    }
+    var major = [];
+    if ($('anytime').checked) {
+        for (var i = 0; i < 2; ++i) {
+            major.push($('at' + i).value);
+        }
+    } else {
+        for (var i = 0; i < 6; ++i) {
+            if ($('major' + i).value) {
+                major.push($('major' + i).value);
+            }
+        }
+    }
+    var flavor = [];
+    for (var i = 0; i < 12; ++i) {
+        if ($('flavor' + i).value) {
+            flavor.push([$('cf'+i+'a').value.toLowerCase(), $('cf'+i+'b').value.toLowerCase(), $('flavor' + i).value]);
+        }
+    }
+	
+	quest.name = $('name').value;
+	quest.xp = $('xp').value;
+	quest.arc = arc;
+	quest.color = $('color').value.toLowerCase();
+	quest.anytimep = $('anytime').checked;
+	quest.major = major;
+	quest.flavor = flavor;
+	quest.desc = $('desc').value;
+}
+
+function translateObjectToForm() {
+	$('name').value = quest.name;
+	$('xp').value = quest.xp;
+	$('color').value = capitalize(quest.color);
+	$('anytime').checked = quest.anytimep;
+	$('desc').value = quest.desc;
+	
+	// Set the arc assignments
+    for (var i = 0; i < 5; ++i) {
+        for (var c = 0; c < 8; ++c) {
+			if ((i in quest.arc) && quest.arc[i].includes(colors[c])) {
+				$(colors[c] + '.' + i).checked = true;
+			} else {	
+				$(colors[c] + '.' + i).checked = false;
+			}
+        }
+    }
+	
+	// Set the major/anytime goals
+	if (quest.anytimep) {
+        for (var i = 0; i < 2; ++i) {
+			if (quest.major[i]) {
+				$('at' + i).value = quest.major[i];
+			} else {
+				$('at' + i).value = "";
+			}
+        }
+	} else {
+        for (var i = 0; i < 6; ++i) {
+			if (quest.major[i]) {
+				$('major' + i).value = quest.major[i];
+			} else {
+				$('major' + i).value = "";
+			}
+        }
+	}
+	
+	// Set the flavors
+    for (var i = 0; i < 12; ++i) {
+        if (quest.flavor[i]) {
+			$('cf'+i+'a').value = capitalize(quest.flavor[i][0]);
+			$('cf'+i+'b').value = capitalize(quest.flavor[i][1]);
+			$('flavor' + i).value = quest.flavor[i][2];
+        } else {
+			$('cf'+i+'a').value = 'Purple';
+			$('cf'+i+'b').value = '';
+			$('flavor' + i).value = '';
+		}
+    }
+	
+	update();
+}
+
+function uploadJSON() {
+	$('importFile').click();
+}
+
+function importJSON() {
+	let files = $('importFile').files;
+	if (!files.length) {
+		alert('No file selected!');
+		return;
+	}
+	
+	var fr = new FileReader();
+
+	fr.onload = function(e) {
+		quest = JSON.parse(e.target.result);
+		translateObjectToForm()
+	}
+
+	fr.readAsText(files[0]);
+	
+	$('importFile').value = ""; // Clear the file after importing it
+}
+
+function exportJSON() {
+	var a = document.createElement("a");
```

Comment:
I'm sure my original spacing was a mess but could we at least try to avoid mixing tabs and spaces? Probably replacing all tabs with four spaces will do the trick.